miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/display/MiniAppWebViewClient.kt

```diff
@@ -25,9 +25,19 @@ internal class MiniAppWebViewClient(
         error: WebResourceError
     ) {
         if (request.url != null && request.url.toString().startsWith(customScheme)) {
-            view.loadUrl(request.url.toString().replace(customScheme, customDomain))
+            loadWithCustomDomain(view, request.url.toString().replace(customScheme, customDomain))
             return
         }
         super.onReceivedError(view, request, error)
     }
+
+    @Suppress("MagicNumber")
+    @VisibleForTesting
+    internal fun loadWithCustomDomain(view: WebView, requestUrl: String) {
+        view.stopLoading()
+        view.postDelayed(
+            { view.loadUrl(requestUrl) },
+            100
```

Comment:
Is `stopLoading` asynchronous? Why was 100ms chosen as the delay? It seems like this still leaves the possibility that `loadUrl` will get called before the view has stopped loading.

I was thinking maybe you could use `onPageFinished` for this. For example, you could call `stopLoading` in `onReceivedError` and then use `onPageFinished` for calling `loadUrl`. Do you think that would work?